crates/bevy_transform/src/hierarchy/hierarchy.rs

```diff
@@ -55,6 +55,17 @@ impl<'w, 's, 'a> DespawnRecursiveExt for EntityCommands<'w, 's, 'a> {
     }
 }
 
+impl<'w> DespawnRecursiveExt for EntityMut<'w> {
+    /// Despawns the provided entity and its children.
+    fn despawn_recursive(&mut self) {
+        let entity = self.id();
+        unsafe {
+            despawn_with_children_recursive(self.world_mut(), entity);
```

Comment:
Could you make the unsafe scope minimal by using instead.                `despawn_with_children_recursive(unsafe { self.world_mut() }, entity);` ?

Reply:
I could, but I prefer to have the unsafe block also cover the part that makes it sound.

Reply:
I reread despawn_with_children_recursive again and it is calling despawning this entity itself too. Calling update_location after that would reveal the error here. There is no location of this entity anymore in this world and its .unwrap() should panic in this case.

Reply:
Ah, of course. This also shows the semantic problem with the trait - I'll update it to take `self` instead.